packages/react/src/useUploadThing.ts

```diff
@@ -131,10 +132,19 @@ export const INTERNAL_uploadthingHookGen = <
       }
     });
 
+    const routeConfig = useRouteConfig(initOpts.url, endpoint as string);
+
     return {
       startUpload,
       isUploading,
-      permittedFileInfo,
+      routeConfig,
+
+      /**
+       * @deprecated Use `routeConfig` instead
+       */
+      permittedFileInfo: routeConfig
+        ? { slug: endpoint, config: routeConfig }
+        : undefined,
```

Comment:
the `permittedFileInfo: { slug: string, config: ExpandedRouteConfig }` is just an "annoyance" - you only care about the config part. the `slug` is what you put into the `useUploadThing(slug, opts)` anyways... Also `routeConfig` is a better name